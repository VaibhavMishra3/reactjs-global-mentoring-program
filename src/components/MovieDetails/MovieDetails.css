.movie-details {
  display: flex;
  background-color: #232323;
  color: #ffffff;
  padding: 20px;
  gap: 30px;
}

.movie-details img {
  max-width: 300px;
  border-radius: 4px;
}

.details-info {
  display: flex;
  flex-direction: column;
}

.details-info h2 {
  font-size: 36px;
  color: #f65261;
  margin: 0 0 10px 0;
}

.details-info p {
  margin: 5px 0;
  font-size: 16px;
  color: #ffffff;
}

.details-info p:first-of-type {
  font-weight: bold;
  font-size: 24px;
  color: #f65261;
}
