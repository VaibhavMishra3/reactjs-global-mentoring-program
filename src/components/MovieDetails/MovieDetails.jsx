import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';

import MoviePoster from '../MoviePoster/MoviePoster';
import { formatGenres, formatYear, formatRuntime } from '../../utils/formatters.js';
import { movieType } from '../../constants/types.js';

const MovieDetails = ({ movie }) => {
  return (
    <Container data-testid='movie-details'>
      <Row>
        <Col md={3}>
          <MoviePoster
            testId={'movie-details-poster'}
            src={movie.poster_path}
            alt={movie.title}
          />
        </Col>
        <Col md={9}>
          <div className='d-flex'>
            <span data-testid='movie-details-title' className='h3'>{movie.title}</span>
            {movie.vote_average && <span data-testid='movie-details-vote' className='h5 ms-4 border border-2 rounded-circle'>{movie.vote_average}</span>}
          </div>
          {movie.genres && <div data-testid='movie-details-genres' className='d-flex mb-4'>{formatGenres(movie.genres)}</div>}
          <div className='d-flex'>
            {movie.release_date && <span data-testid='movie-details-year' className='h5 text-danger'>{formatYear(movie.release_date)}</span>}
            {movie.runtime && <span data-testid='movie-details-runtime' className='h5 ms-4 text-danger'>{formatRuntime(movie.runtime)}</span>}
          </div>
          <div data-testid='movie-details-overview' className='d-flex h5 mt-4 text-start'>{movie.overview}</div>
        </Col>
      </Row>
    </Container>
  )
}

MovieDetails.propTypes = {
  movie: movieType.isRequired
};

export default MovieDetails;
