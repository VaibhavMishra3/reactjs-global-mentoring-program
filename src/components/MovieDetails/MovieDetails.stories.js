import MovieDetails from './MovieDetails';
import { MOCK_MOVIES } from '../../tests/mockData.js';

export default {
  component: MovieDetails
};

export const Default = {
  args: {
    movie: MOCK_MOVIES[0],
  },
  parameters: {
  }
};
