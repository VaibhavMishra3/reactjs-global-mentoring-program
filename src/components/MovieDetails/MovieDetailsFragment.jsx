import React from 'react';
import Link from 'next/link';

import MovieDetails from './MovieDetails';
import { filterType, movieType } from '../../constants/types.js';
import { navigationByFilter } from '../../utils/navigation.js';

const MovieDetailsFragment = ({ filter, movie }) => {
  return (
    <div>
      <Link
        data-testid='close-movie-details'
        className='btn-close float-end'
        href={navigationByFilter('/', filter)}
      />
      <MovieDetails movie={movie} />
    </div>
  );
}

MovieDetailsFragment.propTypes = {
  filter: filterType.isRequired,
  movie: movieType.isRequired
};

export default MovieDetailsFragment;
