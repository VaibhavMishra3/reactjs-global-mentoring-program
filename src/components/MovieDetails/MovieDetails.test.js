import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react'

import MovieDetails from './MovieDetails';
import { MOCK_MOVIES } from '../../constants/mockData.js';

describe('MovieDetails', () => {

  afterEach(cleanup);

  it('should match snapshot', () => {
    const movieDetails = renderer
      .create(
        <MovieDetails
          movie={MOCK_MOVIES[0]}
        />
      )
      .toJSON();

    expect(movieDetails).toMatchSnapshot();
  });

  it('render correctly', () => {
    const movie = MOCK_MOVIES[0];

    render(
      <MovieDetails
        movie={movie}
      />
    );

    const image = screen.getByTestId('movie-details-poster');
    expect(image.getAttribute('src')).toEqual(movie.poster_path);
    expect(image.getAttribute('alt')).toEqual(movie.title);

    expect(screen.getByTestId('movie-details-title')).toHaveTextContent(movie.title);
    expect(screen.getByTestId('movie-details-vote')).toHaveTextContent(movie.vote_average);
    expect(screen.getByTestId('movie-details-genres')).toHaveTextContent('Thriller, Crime');
    expect(screen.getByTestId('movie-details-year')).toHaveTextContent('1994');
    expect(screen.getByTestId('movie-details-runtime')).toHaveTextContent('2h 34min');
    expect(screen.getByTestId('movie-details-overview')).toHaveTextContent(movie.overview);
  });

});
