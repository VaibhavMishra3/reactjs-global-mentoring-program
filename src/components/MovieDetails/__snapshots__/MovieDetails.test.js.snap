// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match snapshot 1`] = `
<div
  className="container"
  data-testid="movie-details"
>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <img
        alt="Pulp Fiction"
        className="img-fluid"
        data-nimg="1"
        data-testid="movie-details-poster"
        decoding="async"
        height={500}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FdM2w364MScsjFf8pfMbaWUcWrR.jpg&w=640&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FdM2w364MScsjFf8pfMbaWUcWrR.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw500%2FdM2w364MScsjFf8pfMbaWUcWrR.jpg&w=640&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={250}
      />
    </div>
    <div
      className="col-md-9"
    >
      <div
        className="d-flex"
      >
        <span
          className="h3"
          data-testid="movie-details-title"
        >
          Pulp Fiction
        </span>
        <span
          className="h5 ms-4 border border-2 rounded-circle"
          data-testid="movie-details-vote"
        >
          8.3
        </span>
      </div>
      <div
        className="d-flex mb-4"
        data-testid="movie-details-genres"
      >
        Thriller, Crime
      </div>
      <div
        className="d-flex"
      >
        <span
          className="h5 text-danger"
          data-testid="movie-details-year"
        >
          1994
        </span>
        <span
          className="h5 ms-4 text-danger"
          data-testid="movie-details-runtime"
        >
          2h 34min
        </span>
      </div>
      <div
        className="d-flex h5 mt-4 text-start"
        data-testid="movie-details-overview"
      >
        A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.
      </div>
    </div>
  </div>
</div>
`;
